datafusion_iceberg/src/table.rs

```diff
@@ -184,7 +186,9 @@ impl TableProvider for DataFusionTable {
                 };
                 let statement = DFParserBuilder::new(sql).build()?.parse_statement()?;
                 let logical_plan = session_state.statement_to_plan(statement).await?;
-                ViewTable::try_new(logical_plan, Some(sql.clone()))?
+                let logical_plan = Analyzer::with_rules(vec![Arc::new(TypeCoercion::new())])
+                    .execute_and_check(logical_plan, session.config_options(), |_, _| {})?;
+                ViewTable::new(logical_plan, Some(sql.clone()))
```

Comment:
`ViewTable::try_new` is deprecated, and it told me to apply TypeCoercion manually - whether that is actually required in this case, I'm not sure about.